src/base/name/builder.rs

```diff
@@ -648,68 +625,79 @@ impl From<PushError> for FromStrError {
 impl From<PushNameError> for FromStrError {
     fn from(err: PushNameError) -> FromStrError {
         match err {
-            PushNameError::LongName => FromStrError::LongName,
+            PushNameError::LongName => PresentationErrorEnum::LongName.into(),
             PushNameError::ShortBuf => FromStrError::ShortBuf,
         }
     }
 }
 
-impl From<LabelFromStrError> for FromStrError {
-    fn from(err: LabelFromStrError) -> FromStrError {
-        match err {
-            LabelFromStrError::UnexpectedEnd => FromStrError::UnexpectedEnd,
-            LabelFromStrError::BinaryLabel => FromStrError::BinaryLabel,
-            LabelFromStrError::LongLabel => FromStrError::LongLabel,
-            LabelFromStrError::IllegalEscape => FromStrError::IllegalEscape,
-            LabelFromStrError::IllegalCharacter(ch) => {
-                FromStrError::IllegalCharacter(ch)
-            }
-        }
+impl<T: Into<PresentationError>> From<T> for FromStrError {
+    fn from(err: T) -> Self {
+        Self::Presentation(err.into())
     }
 }
 
-impl From<SymbolCharsError> for FromStrError {
-    fn from(err: SymbolCharsError) -> FromStrError {
-        use crate::base::scan::SymbolCharsEnum;
+//--- Display and Error
 
-        match err.0 {
-            SymbolCharsEnum::BadEscape => Self::IllegalEscape,
-            SymbolCharsEnum::ShortInput => Self::UnexpectedEnd,
+impl fmt::Display for FromStrError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match *self {
+            FromStrError::Presentation(err) => err.fmt(f),
+            FromStrError::ShortBuf => ShortBuf.fmt(f),
         }
     }
 }
 
+#[cfg(feature = "std")]
+impl std::error::Error for FromStrError {}
+
+//------------ PresentationError ---------------------------------------------
+
+/// An illegal presentation format was encountered.
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub struct PresentationError(PresentationErrorEnum);
+
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+enum PresentationErrorEnum {
+    BadLabel(LabelFromStrError),
```

Comment:
This would be more consistent with examples earlier in the review if the inner type were called `BadLabelError` - was it intentionally named differently?

Reply:
`LabelFromStrError` is a public type used by `Label`s `FromStr` impl, so I think the name makes sense. The variant name makes sense to me as it indicates that a label was rejected.